src/plugins/ccode/testmod_ccode.c

```diff
@@ -51,48 +39,53 @@ void test_encode (void)
 {
 	printf ("test encode\n");
 
-	CCodeData * d = get_data ();
+	CCodeData * mapping = get_data ();
 
-	char buf[1000];
-	d->buf = buf;
+	unsigned char buffer[1000];
+	mapping->buffer = buffer;
 	Key * test = keyNew ("user/test", KEY_VALUE, decoded_string, KEY_END);
-	elektraCcodeEncode (test, d);
-	succeed_if (!memcmp (keyValue (test), encoded_string, sizeof (encoded_string) - 1), "string not correctly encoded");
+	encodeKey (test, mapping);
+	succeed_if (memcmp (keyValue (test), encoded_string, sizeof (encoded_string) - 1) == 0, "string not correctly encoded");
 
-	elektraFree (d);
+	elektraFree (mapping);
 	keyDel (test);
 }
 
 void test_decode (void)
 {
 	printf ("test decode\n");
 
-	CCodeData * d = get_data ();
+	CCodeData * mapping = get_data ();
 
-	char buf[1000];
-	d->buf = buf;
+	unsigned char buffer[1000];
+	mapping->buffer = buffer;
 	Key * test = keyNew ("user/test", KEY_SIZE, sizeof (encoded_string) - 1, KEY_VALUE, encoded_string, KEY_END);
-	elektraCcodeDecode (test, d);
-	succeed_if (!memcmp (keyValue (test), decoded_string, sizeof (decoded_string) - 1), "string not correctly encoded");
+	decodeKey (test, mapping);
+	succeed_if (memcmp (keyValue (test), decoded_string, sizeof (decoded_string) - 1) == 0, "string not correctly encoded");
 
-	elektraFree (d);
+	elektraFree (mapping);
 	keyDel (test);
 }
 
 void check_reversibility (const char * msg)
+#ifdef __llvm__
+	__attribute__ ((annotate ("oclint:suppress[deep nested block]"), annotate ("oclint:suppress[high cyclomatic complexity]"),
+			annotate ("oclint:suppress[high ncss method]")))
+#endif
+
 {
-	CCodeData * d = get_data ();
-	char buf[1000];
-	d->buf = buf;
+	CCodeData * mapping = get_data ();
```

Comment:
Does this function test every possible character?

For maintenance it would be nice if the check_reversibility can be shared between the different code plugins.

Reply:
> Does this function test every possible character?

Nope, not as far as I can tell. I [did not write this function](https://github.com/ElektraInitiative/libelektra/commit/0f8f033c) though 😊.

Reply:
Yeah, seems like I was in a hurry finishing my master thesis :smile: